File: HomeBudget.Data/Entities/Account.cs
using HomeBudget.Data.Enums;

namespace HomeBudget.Data.Entities;

public class Account
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public AccountType Type { get; set; }
    public decimal Balance { get; set; }
    public string Color { get; set; } = "#6366F1";
    public string Icon { get; set; } = "account-balance";
    public int UserId { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public User User { get; set; } = null!;
    public ICollection<Transaction> Transactions { get; set; } = new List<Transaction>();
}

File: HomeBudgetMobile/src/models/account.ts
import { AccountType } from "./enums";

export interface AccountDto {
  id: number;
  name: string;
  type: AccountType;
  balance: number;
  color: string;
  icon: string;
  userId: number;
  userName: string;
}

export interface CreateAccountDto {
  name: string;
  type: AccountType;
  balance: number;
  color: string;
  icon: string;
  userId: number;
}

export interface UpdateAccountDto {
  id: number;
  name: string;
  type: AccountType;
  balance: number;
  color: string;
  icon: string;
}
